### ISSUE TITLE
Gradient Accumulation with Custom fit in tf.keras?

### ISSUE THREAD

**Issue by innat** [tensorflow/tensorflow#47578]
## Documents: 
[Customizing fit : TF.Keras](https://keras.io/guides/customizing_what_happens_in_fit/)
This a great article and one of the best deep learning practices. However, with this reading intuition, I was trying to implement Gradiant Accumulation (**GA**) by customizing the `.fit()` method, overriding the `train_step` within `tf.keras.Model` class. But I couldn't implement it yet. 

I don't want it in a custom training loop (like [this](https://stackoverflow.com/questions/59893850/how-to-accumulate-gradients-in-tensorflow-2-0)) but with customized `.fit()`. I also asked in [SO](https://stackoverflow.com/q/66472201/9215780), no response yet. I've found some workaround [here](https://github.com/keras-team/keras/issues/14483), but it's too messy. There should be a convenient way to achieve this. How to achieve this? Please find the gist [here](https://colab.research.google.com/drive/1hr9kIhTlkCxlW_md3CXWREor-6TE6RFq?usp=sharing). 

Also, I know the pros of using **GA** but what are the major **cons** of using it? Why it's not come as a default but an optional feature with the framework?

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3, TF v2.4 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/88e42704e1f5f9f9ed14766013e25e0e/47578.ipynb). Thanks!

**Comment by innat** [comment]
Any effective workaround?

**Comment by innat** [comment]
I've done it. Here is the [gist](https://gist.github.com/innat/985b10357f7a33806308e3147c78c823) and more [discussion](https://stackoverflow.com/a/66512358/9215780). Closing this issue.

**Comment by innat** [comment]
@jvishnuvardhan this issue is not yet solved. Please re-open the issue.
